L1Trigger/L1TCalorimeter/plugins/L1TStage2InputPatternWriter.cc

```diff
@@ -265,53 +271,82 @@ void
 L1TStage2InputPatternWriter::endJob() 
 {
 
-  LogDebug("L1TDebug") << "Read " << nFrame_ << " frames" << std::endl;
+  //frames per event
+  unsigned int framesPerEv = nHeaderFrames_ + nPayloadFrames_ + nClearFrames_;
 
-  // write file
-  std::ofstream file( filename_ );
+  //events per file
+  unsigned int evPerFile = floor(1024/framesPerEv);
 
-  file << "Board MP7_TEST" << std::endl;
+  //frames per file
+  unsigned int framesPerFile = framesPerEv*evPerFile;
 
-  // quad/chan numbers
-  file << " Quad/Chan : ";
-  for ( unsigned i=0; i<nQuad_; ++i ) {
-    for ( unsigned j=0; j<nChan_; ++j ) {
-      file << "   q" << i << "c" << j << "   ";
-    }
-  }
-  file << std::endl;
+  //number of output files
+  unsigned int nOutFiles = ceil(nEvents_/evPerFile);
 
-  // link numbers
-  file << "      Link : ";
-  for ( unsigned i=0; i<nQuad_; ++i ) {
-    for ( unsigned j=0; j<nChan_; ++j ) {
-      file << "    " << (i*nChan_)+j << "       ";
+  LogDebug("L1TDebug") << "Read " << nFrame_ << " frames" << std::endl;
+  LogDebug("L1TDebug") << "Read " << nEvents_ << " events" << std::endl;
+  LogDebug("L1TDebug") << "Writing " << nOutFiles << " files" << std::endl;
+  LogDebug("L1TDebug") << "Output directory: ./" << outDir_ << "/" << std::endl;
+  
+  //files
+  std::vector< std::ofstream > outFiles(nOutFiles);
+    
+  //make output files and write to them
+  for(uint itFile=0; itFile<nOutFiles; ++itFile){
+    std::stringstream outFilename;
```

Comment:
This is not used in any of the workflows.  This module is used by experts when they build test files.